clang/test/SemaHLSL/VectorOverloadResolution.hlsl

```diff
@@ -1,5 +1,8 @@
 // RUN: %clang_cc1 -triple dxil-unknown-shadermodel6.6-library -S -fnative-half-type -finclude-default-header -o - -ast-dump %s | FileCheck %s
 // RUN: %clang_cc1 -finclude-default-header -triple dxil-pc-shadermodel6.6-library %s -fnative-half-type -emit-llvm -disable-llvm-passes -o - | FileCheck %s --check-prefixes=CHECKIR
+
+export {
```

Comment:
So I think there was a choice here between marking each individual function as `export` or using the big block around the file.  I think there may also be a third option involving a command-line parameter?

Anyway, I think we're ending up now with the tests being inconsistent - there's some individually marked and there's some marked with the bock. I think the choice was mainly made to save typing? Are we confident it's the right one?

Reply:
Marking the individual functions as `export` or using the big `export` block around the file is equivalent. I don't think the inconsistent use in the test is an issue. Each test author can pick a style that works best for each test. If there is just a single or a couple of functions it seems easier to mark each function, and if there are more then an `export` block makes more sense. Another option is to include a shader point function that calls all the other functions to make sure the code gets generated. 

Unlike DXC we do not yet have a command line option to change the default linkage ([hlsl-specs/issues#242](https://github.com/microsoft/hlsl-specs/issues/242)) and I don't think Clang's `-fvisibility` applies here.